File: chap6-cross-paradigm/1-object-as-resource-owner/cpp/main.cpp

#include<iostream>
#include "file.cpp"

int main() {
    File f1("file1.txt");
    std::cout << f1.fd << "\n";

    File f_copy = f1; // Implicit copy constructor
    File f_move = std::move(f1); // Move constructor

    // Use moved object
    std::cout << f_move.fd << "\n";
}

File: chap6-cross-paradigm/1-object-as-resource-owner/src/main.rs
mod file;
use crate::file::File;

fn main() {
    let mut f1 = File::new("file1.txt");

    let f_copy = f1.clone(); // Copy emulated
    let f_move = f1; // Implicit move

    // Use moved object
    println!("{}", f_move.fd);
}
